addons/hr_timesheet/controllers/portal.py

```diff
@@ -58,7 +58,10 @@ def _get_search_domain(self, search_in, search):
         if search_in in ('task', 'all'):
             search_domain = OR([search_domain, [('task_id', 'ilike', search)]])
         if search_in == 'parent_task_id':
-            search_domain = OR([search_domain, [('parent_task_id', '=', int(search))]])
+            parent_task_id = int(search)
+            sub_task_ids = request.env['project.task'].browse(parent_task_id)._get_subtask_ids_per_task_id()[parent_task_id]
+            sub_task_ids.append(parent_task_id)
+            search_domain = OR([search_domain, [('parent_task_id', 'in', sub_task_ids)]])
```

Comment:
We should check if `parent_task_id` is not falsy otherwise the dict returned by `_get_subtask_ids_per_task_id` will be empty and it will crash because we don't find the key inside the dict. 😬 

I guess we will have something like: 

```suggestion
            if parent_task_id:
                sub_task_ids = request.env['project.task'].browse(parent_task_id)._get_subtask_ids_per_task_id()[parent_task_id]
                sub_task_ids.append(sub_task_id)
                search_domain = OR([search_domain, [('parent_task_id', 'in', sub_task_ids)]])
```

(I did not test my suggestion)